### PR TITLE
[New] use `process.env.NODE_TAPE_OBJECT_PRINT_DEPTH` for the default object print depth

### PR DESCRIPTION
**Pull request by inadarei**
Issue #419

### CODE REVIEW

File: readme.markdown [hunk 1]
```diff
@@ -15,10 +15,10 @@ var test = require('tape');
 
 test('timing test', function (t) {
     t.plan(2);
-
```
Comment (ljharb) [inline readme.markdown]: please revert unrelated whitespace changes.

Comment (inadarei) [reply readme.markdown]: Ha, yeah, I was just looking into how to stop VS Code from adding those :)

File: readme.markdown [hunk 2]
```diff
@@ -148,7 +148,8 @@ Available `opts` options are:
 - opts.timeout = 500. Set a timeout for the test, after which it will fail.
   See test.timeoutAfter.
 - opts.objectPrintDepth = 5. Configure max depth of expected / actual object
-  printing.
+  printing. Environmental variable `NODE_TAPE_OBJECT_PRINT_DEPTH` can set the 
+  desired depth for all tests and overrides any locally-set values.
```
Comment (ljharb) [inline readme.markdown]: should this override? or should the env var merely set the default, so that local intentional overrides still take precedence?

Comment (inadarei) [reply readme.markdown]: Was wondering the same, since good arguments can be made for both approaches. Finally decided to go the current route because:
1. seems more compatible with 12-Factor Apps guidance
2. env vars overriding is how https://www.npmjs.com/package/config works

Comment (ljharb) [reply readme.markdown]: I've no interest in complying with "12-factor apps" by itself; if it isn't the best thing for the project, then it shouldn't be done.

In general, i would agree that an env var should always override an *equivalent setting* - but there is no equivalent for "set every object print depth" in tape. Currently, every one is per-test, and per-test overrides imo should always take precedence.

Comment (inadarei) [reply readme.markdown]: OK, no problem. That makes sense. I was just explaining the logic behind the initial decision, but not doing it that way also makes sense. I will make the change as you requested.

File: lib/test.js [hunk 3]
```diff
@@ -52,11 +52,22 @@ function Test (name_, opts_, cb_) {
...
     this._skip = args.opts.skip || false;
     this._timeout = args.opts.timeout;
-    this._objectPrintDepth = args.opts.objectPrintDepth || 5;
     this._plan = undefined;
     this._cb = args.cb;
     this._progeny = [];
     this._ok = true;
+    var depthEnvVar = 'NODE_TAPE_OBJECT_PRINT_DEPTH';
+    if (args.opts.objectPrintDepth) {
+        this._objectPrintDepth = args.opts.objectPrintDepth
```
Comment (ljharb) [inline lib/test.js]: this line is missing a semicolon

File: lib/test.js [hunk 4]
```diff
@@ -52,11 +52,22 @@ function Test (name_, opts_, cb_) {
...
+    } else if (process.env[depthEnvVar]) {
```
Comment (ljharb) [inline lib/test.js]: instead of storing the key name in a variable, and extracting it multiple times, can we just extract the value into a variable right off the bat?

File: lib/test.js [hunk 5]
```diff
@@ -52,11 +52,22 @@ function Test (name_, opts_, cb_) {
...
+        if (process.env[depthEnvVar].toLowerCase() === 'infinity') {
```
Comment (ljharb) [inline lib/test.js]: we should try to be robust against `delete String.prototype.toLowerCase && delete RegExp.prototype.test` (by caching the prototype method at module level)

File: lib/test.js [hunk 6]
```diff
@@ -52,11 +52,22 @@ function Test (name_, opts_, cb_) {
...
+            this._objectPrintDepth = Infinity;
+        } else {
+            this._objectPrintDepth = parseInt(process.env[depthEnvVar], 10);
```
Comment (ljharb) [inline lib/test.js]: hm, i'm not actually sure about the parseInt here. tape doesn't seem to do any typechecking on the `objectPrintDepth` option - ie, you could pass a non-number or a non-integer and it'd just coerce to a number later. Maybe we should do the same here?

Comment (inadarei) [reply lib/test.js]: ok

File: lib/test.js (hunk 5 above)
Comment (inadarei) [reply lib/test.js]: Sorry, I am not sure I understand what you are proposing.

File: lib/test.js [hunk 7]
```diff
@@ -56,14 +56,14 @@ function Test (name_, opts_, cb_) {
     this._cb = args.cb;
     this._progeny = [];
     this._ok = true;
-    var depthEnvVar = 'NODE_TAPE_OBJECT_PRINT_DEPTH';
+    var depthEnvVar = process.env['NODE_TAPE_OBJECT_PRINT_DEPTH']
```
Comment (ljharb) [inline lib/test.js]: please always use dot access instead of bracket access in javascript wherever possible.

File: lib/test.js [hunk 8]
```diff
@@ -52,11 +52,22 @@ function Test (name_, opts_, cb_) {
...
-    this._objectPrintDepth = args.opts.objectPrintDepth || 5;
     this._plan = undefined;
     this._cb = args.cb;
     this._progeny = [];
     this._ok = true;
+    var depthEnvVar = process.env.NODE_TAPE_OBJECT_PRINT_DEPTH;
+    if (args.opts.objectPrintDepth) {
+        this._objectPrintDepth = args.opts.objectPrintDepth;
+    } else if (depthEnvVar) {
+        if (depthEnvVar.toLowerCase() === 'infinity') {
```
Comment (ljharb) [inline lib/test.js]: To clarify: we should have `var toLowerCase = String.prototype.toLowerCase;` near the top of the file, and then here, do `toLowerCase.call(depthEnvVar)` instead of assuming that nobody's done `delete String.prototype.toLowerCase` inside their tests.

**Comment by inadarei** [comment]
Thanks for the approval and assitance in getting there. What are the next steps for a merge?
